pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -254,14 +254,20 @@ public SuccessResponse deleteTable(
 
     List<String> tablesDeleted = new LinkedList<>();
     try {
-      if (tableType != TableType.REALTIME && !TableNameBuilder.REALTIME.tableHasTypeSuffix(tableName)) {
+      if (tableType != TableType.REALTIME && !TableNameBuilder.REALTIME.tableHasTypeSuffix(tableName)
+          && _pinotHelixResourceManager.hasOfflineTable(tableName)) {
         _pinotHelixResourceManager.deleteOfflineTable(tableName);
         tablesDeleted.add(TableNameBuilder.OFFLINE.tableNameWithType(tableName));
       }
-      if (tableType != TableType.OFFLINE && !TableNameBuilder.OFFLINE.tableHasTypeSuffix(tableName)) {
+      if (tableType != TableType.OFFLINE && !TableNameBuilder.OFFLINE.tableHasTypeSuffix(tableName)
+          && _pinotHelixResourceManager.hasRealtimeTable(tableName)) {
         _pinotHelixResourceManager.deleteRealtimeTable(tableName);
         tablesDeleted.add(TableNameBuilder.REALTIME.tableNameWithType(tableName));
       }
+      if (tablesDeleted.size() == 0) {
```

Comment:
Right now if the table does not exist, the rest end point still returns "Success [table_non_existent] deleted". This is not consistent with system state -- that is why I think it is a bug (though minor).

Thank you for the overflow. It is interesting and new to me. Both sides has valid points. I think there is argument for idem-potency there -- but Pinot controller is not a stateless service (it has zk data for table state) IMO.

Reply:
I think it would be better to think of this change as an improvement to the pinot admin API feedback. set aside the debates over the correct way to respond to a DELETE on a resource that is not presented, This change provides the caller of this method more information by providing feedback if such method doesn't exist. If the users desire the old behavior, he/she can just check if the method returns non-500 return codes. But now our users are also able to check if the table is really in pinot system. Given how this change improves the existing pinot system behavior, I think we should accept this diff after fixing any further minor issues @Jackie-Jiang